### PR TITLE
Separate training on population from application of ancestry adjustment in prs pipeline

### PR DESCRIPTION
**Pull request by kachulis**
Splits the `AdjustScores`  task into two tasks, `TrainAncestryModel` which trains the ancestry model (shocking), and `AdjustScores`, which applies the trained ancestry model to adjust individual scores.  These tasks (along with a few others) have been moved into a new `PRSTasks.wdl` file, to improve code organization.  The prs scoring wdl can now be passed either a population vcf, in which case it trains and applies the ancestry correction model, or can be passed a pretrained model.

### CODE REVIEW

File: ImputationPipeline/Validation/ValidateScoring.wdl [hunk 1]
```diff
@@ -2,6 +2,11 @@ version 1.0
 
 import "../ScoringPart.wdl" as Scoring
 import "https://raw.githubusercontent.com/broadinstitute/palantir-workflows/main/ImputationPipeline/ScoringPart.wdl" as ScoringMain
```
Comment (michaelgatzen) [inline ImputationPipeline/Validation/ValidateScoring.wdl:4]: Why no relative import?

Comment (kachulis) [reply ImputationPipeline/Validation/ValidateScoring.wdl:4]: we specifically want to import the `main` branch version so we can compare it to whatever version is on "this" branch.  Obviously this is a little silly if you are running it on the main branch, but the intended use is really to test out other branches before merging.  I would love to find a better way to do this type of branch comparison in the future.

Comment (kachulis) [reply ImputationPipeline/Validation/ValidateScoring.wdl:4]: we specifically want to import the main branch version so we can compare it to whatever version is on "this" branch. Obviously this is a little silly if you are running it on the main branch, but the intended use is really to test out other branches before merging. I would love to find a better way to do this type of branch comparison in the future.

File: ImputationPipeline/Validation/ValidateScoring.wdl [hunk 2]
```diff
@@ -277,34 +290,35 @@ task CompareRawScores {
...
 
-		array_scores <- read_tsv("~{arrayScores}") %>% transmute(`#IID`, raw_score_array=SCORE1_SUM)
-		wgs_score <- read_tsv("~{wgsScores}") %>% transmute(`#IID`, raw_score_wgs=SCORE1_SUM)
+		x_scores <- read_tsv("~{x_axis_scores}") %>% transmute(IID, adjusted_score_x=adjusted_score)
+		y_scores <- read_tsv("~{y_axis_scores}") %>% transmute(IID, adjusted_score_y=adjusted_score)
 
-		sample_names <- read_delim("~{sample_name_map}", delim=":", col_names=FALSE)
+		sample_names_map <- if(~{true="TRUE" false="FALSE" defined(sample_name_map)}) read_delim("~{sample_name_map}", delim=":", col_names=FALSE) else x_scores %>% transmute(X1=IID, X2=IID)
 
+		combined_scores <- inner_join(inner_join(x_scores, sample_names_map, by=c("IID"="X1")), y_scores, by=c("X2"="IID"))
```
[the remark this answers is not in the document]
Comment (kachulis) [reply ImputationPipeline/Validation/ValidateScoring.wdl]: Yes, using more dplyr in the score adjusting task leads to this.

[the remark this answers is not in the document]
Comment (kachulis) [reply ImputationPipeline/Validation/ValidateScoring.wdl]: I think it happens at the join at https://github.com/broadinstitute/palantir-workflows/pull/91/files#diff-a6c9d5ef41650d27cfed259ca8e0cea6e6e4ea8d6b5c6bd14b1a8ec4505221d5R386

File: ImputationPipeline/Validation/ValidateScoring.wdl (hunk 1 above)
Comment (michaelgatzen) [reply ImputationPipeline/Validation/ValidateScoring.wdl:4]: Oh, that makes a lot of sense

**Review by michaelgatzen — APPROVED** [review]

**Pull request merged**